.pre-commit-config.yaml

```diff
@@ -0,0 +1,135 @@
+---
+exclude: ^(subprojects|third-party)/
+default_language_version:
+  # force all unspecified python hooks to run python3
+  python: python3
+
+repos:
+# C source format; minor difference between .c / .h, much different style for the fio io-engine.
+# For the details, then compare: scripts/clang-format-*
+- repo: local
+  hooks:
+  - id: clang-format
+    name: C-format-headers
+    language: system
+    entry: scripts/pcf_clang_format.py
+    args: ["--style-file", "scripts/clang-format-h"]
+    files: \.h$
+
+  - id: clang-format
+    name: C-format-source
+    language: system
+    entry: scripts/pcf_clang_format.py
+    args: ["--style-file", "scripts/clang-format-c"]
+    files: \.c$
+    exclude: xnvme_fioe.c$
+
+  - id: clang-format
+    name: C-format-fioe
+    language: system
+    entry: scripts/pcf_clang_format.py
+    args: ["--style-file", "scripts/clang-format-fioe"]
+    files: xnvme_fioe\.c$
+
+# Git: intended for commit-message formating
+- repo: local
+  hooks:
+  - id: commit-msg-check
+    name: Git-Validate commit message
+    language: pygrep
+    entry: '\A(?![a-z]+:[a-z]+: .+\nSigned-off-by:.+)'
+    args: [--multiline]
+    stages: [commit-msg]
+
+# Yaml: intended for the GitHUB Actions .yaml
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.1.0
+  hooks:
+  - id: check-yaml # pre-commit-hook-yamlfmt only works on valid YAML files
+    name: YAML-format-check
+- repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt
+  rev: 0.1.0
+  hooks:
+  - id: yamlfmt
+    name: YAML-format
+    args:
+    - '--mapping=2'
+    - '--sequence=2'
+    - '--offset=0'
+    - '--width=120'
+    - '--preserve-quotes'
+    types: [yaml]
+
+# Python: for all the helpers in scripts/*
+- repo: https://github.com/psf/black
+  rev: '22.1.0'
```

Comment:
Should be updated to `'22.3.0'`. This fixes a bug which is blocking CI right now

Reply:
https://github.com/psf/black/issues/2964

Reply:
Fixed